lib/rspec_profiling/current_commit.rb

```diff
@@ -4,24 +4,35 @@ module RspecProfiling
   module CurrentCommit
     extend self
 
-    def git?
+    def version_control
```

Comment:
Rather than repeating the case logic below, I think we'd rather extract classes for each version control type that implement the correct `sha` and `time` methods and instantiate the correct one here.